pkg/diff/sql_generator.go

```diff
@@ -1461,6 +1514,75 @@ func (s sequenceOwnershipSQLVertexGenerator) GetDeleteDependencies(_ schema.Sequ
 	return nil
 }
 
+type extensionSQLVertexGenerator struct{}
+
+func (e *extensionSQLVertexGenerator) Add(extension schema.Extension) ([]Statement, error) {
+	s := fmt.Sprintf("CREATE EXTENSION %s", extension.GetName())
+	if len(extension.Version) != 0 {
+		s += fmt.Sprintf(" WITH VERSION '%s'", extension.Version)
+	}
+	return []Statement{{
+		DDL:     s,
+		Timeout: statementTimeoutDefault,
+		Hazards: nil,
+	}}, nil
+}
+
+func (e *extensionSQLVertexGenerator) Delete(extension schema.Extension) ([]Statement, error) {
+	return []Statement{{
+		DDL:     fmt.Sprintf("DROP EXTENSION %s", extension.GetName()),
+		Timeout: statementTimeoutDefault,
+		Hazards: []MigrationHazard{migrationHazardExtensionDroppedCannotTrackDependencies},
+	}}, nil
+}
+
+func (e *extensionSQLVertexGenerator) Alter(diff extensionDiff) ([]Statement, error) {
+	var statements []Statement
+	if diff.new.Version != diff.old.Version {
+		if len(diff.new.Version) == 0 {
+			// This is an implicit upgrade to the latest extension version.
+			statements = append(statements, Statement{
+				DDL:     fmt.Sprintf("ALTER EXTENSION %s UPDATE", diff.new.Name),
+				Timeout: statementTimeoutDefault,
+				Hazards: []MigrationHazard{migrationHazardExtensionAlteredVersionUpgraded},
+			})
+		} else {
+			oldVersion, err := strconv.ParseFloat(diff.old.Version, 64)
+			if err != nil {
+				return nil, fmt.Errorf("parsing extension version to float: %w", err)
+			}
+
+			newVersion, err := strconv.ParseFloat(diff.new.Version, 64)
+			if err != nil {
+				return nil, fmt.Errorf("parsing extension version to float: %w", err)
+			}
+
+			if newVersion < oldVersion {
+				return nil, fmt.Errorf("new extension version is lower than the old one")
+			}
+			statements = append(statements, Statement{
```

Comment:
Postgres extension versions are not necessarily numbers. They can be strings like `1.3.2`. I think we shouldn't ban downgrading either (to at least avoid this parsing logic)